include/static_math/trigonometry.inl

```diff
@@ -0,0 +1,317 @@
+/*
+ * Copyright (C) 2013-2015 Morwenn
+ *
+ * static_math is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version.
+ *
+ * static_math is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program. If not,
+ * see <http://www.gnu.org/licenses/>.
+ */
+
+ /*
+ * This file is the work of theLOLflashlight's GitHub account.
+ */
+
+////////////////////////////////////////////////////////////
+// Helper functions
+
+namespace details
+{
+    constexpr size_t EXP_MAX_DEPTH = 50;
+    constexpr size_t SIN_MAX_DEPTH = 51;
+    constexpr size_t COS_MAX_DEPTH = 50;
+
+    static_assert(smath::is_odd(SIN_MAX_DEPTH), "");
+    static_assert(smath::is_even(COS_MAX_DEPTH), "");
```

Comment:
Are these assertions realy needed? I'm all for testing and the other assertions in the header look useful, but those two are hard to get wrong. I think we can do without them.